core/src/avm2/globals/flash/text/engine/text_block.rs

```diff
@@ -0,0 +1,147 @@
+use crate::avm2::activation::Activation;
+use crate::avm2::error::Error;
+use crate::avm2::globals::flash::display::display_object::initialize_for_allocator;
+use crate::avm2::object::{Object, TObject};
+use crate::avm2::parameters::ParametersExt;
+use crate::avm2::value::Value;
+use crate::avm2::Multiname;
+use crate::avm2_stub_method;
+use crate::display_object::EditText;
+use crate::html::TextFormat;
+use crate::string::WStr;
+
+pub fn create_text_line<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    avm2_stub_method!(activation, "flash.text.TextBlock", "createTextLine");
+
+    let previous_text_line = args.try_get_object(activation, 0);
+    let width = args.get_f64(activation, 1)?;
+
+    let content = this.get_public_property("content", activation)?;
+
+    let content = if matches!(content, Value::Null) {
+        return Ok(Value::Null);
+    } else {
+        content.as_object().unwrap()
+    };
+
+    let text = match previous_text_line {
+        Some(_) => {
+            // Some SWFs rely on eventually getting `null` from createLineText.
+            // TODO: Support multiple lines
+            this.set_property(
+                &Multiname::new(
+                    activation.avm2().flash_text_engine_internal,
+                    "_textLineCreationResult",
+                ),
+                "complete".into(),
+                activation,
+            )?;
+            return Ok(Value::Null);
+        }
+        // Get the content element's text property.
+        // TODO: GraphicElement and GroupElement
+        None => content
+            .get_public_property("text", activation)
+            .and_then(|o| {
+                if matches!(o, Value::Null) {
+                    Ok("".into())
+                } else {
+                    o.coerce_to_string(activation)
+                }
+            })
+            .unwrap_or("".into()),
+    };
+
+    let class = activation.avm2().classes().textline;
+    let movie = activation.context.swf.clone();
+
+    // FIXME: TextLine should be its own DisplayObject
+    let display_object: EditText =
+        EditText::new_tlf(&mut activation.context, movie, 0.0, 0.0, width, 15.0);
+
+    display_object.set_text(text.as_wstr(), &mut activation.context);
+
+    let element_format = content
+        .get_public_property("elementFormat", activation)?
+        .as_object();
+
+    apply_format(activation, display_object, text.as_wstr(), element_format)?;
+
+    let instance = initialize_for_allocator(activation, display_object.into(), class)?;
+    class.call_native_init(instance.into(), &[], activation)?;
```

Comment:
BTW I don't think `call_native_init` actually does anything here, right?

Reply:
No, I included it because I saw other code do the same thing, but this feels more correct, since we should be calling the `native_init` when constructing a class. If we ever actually do something in the `native_init`, this will work correctly.